components/menu/menu_item.tsx

```diff
@@ -0,0 +1,208 @@
+// See LICENSE.txt for license information.
+
+import React, {ReactElement, ReactNode, Children} from 'react';
+import {styled} from '@mui/material/styles';
+import MuiMenuItem from '@mui/material/MenuItem';
+import type {MenuItemProps as MuiMenuItemProps} from '@mui/material/MenuItem';
+
+export interface Props extends MuiMenuItemProps {
+
+    /**
+     * To support quick recognition of menu item. Could be icon, avatar or emoji.
+     */
+    leadingElement?: ReactNode;
+
+    /**
+     * There can be two labels for a menu item - primaryLabel and secondaryLabel.
+     * If only one element is passed, it will be primaryLabel. And if another element is passed, it will be secondaryLabel.
+     * @example
+     * <Menu.Item labels={<FormattedMessage id="primary.label" defaultMessage="primary Label"/>}/>
+     *
+     * @example
+     * <Menu.Item labels={
+     *  <>
+     *   <FormattedMessage id="primary.label" defaultMessage="primary Label"/>
+     *   <FormattedMessage id="secondary.label" defaultMessage="secondary Label"/>
+     * </>
+     * }/>
+     *
+     * @note
+     * Wraps the labels with element such as span, div etc. to support styling instead of passing text node directly.
+     */
+    labels: ReactElement;
+
+    /**
+     * The meta data element to display extra information about menu item. Could be chevron, shortcut or badge.
+     * It is formed with subMenuDetail and trailingElement. If only one is passed, it will be tailingElement. If two are
+     * passed, first will be subMenuDetail and second will be trailingElement.
+     *
+     * @example
+     * <Menu.Item trailingElements={<ChevronRightIcon/>}/>
+     *
+     * @example
+     * <Menu.Item trailingElements={
+     *  <>
+     *   <FormattedMessage id="submenu.detail" defaultMessage="submenu detail"/>
+     *   <ChevronRightIcon/>
+     * </>
+     * }/>
+     */
+    trailingElements?: ReactNode;
+
+    /**
+     * For actions of menu item that are destructive in nature and harder to undo.
+     */
+    isDestructive?: boolean;
+
+    /**
+     * ONLY to support submenus. Avoid passing children to this component. Support for children is only added to support submenus.
+     */
+    children?: ReactNode;
+}
+
+/**
+ * To be used as a child of Menu component.
+ * Checkout Compass's Menu Item(compass.mattermost.com)  for terminology, styling and usage guidelines.
+ *
+ * @example
+ * <Menu.Container>
+ *    <Menu.Item/>
+ * </Menu.Container>
+ */
+export function MenuItem(props: Props) {
+    const {
+        leadingElement,
+        labels,
+        trailingElements,
+        isDestructive,
+        children,
+        ...restProps
+    } = props;
+
+    // When both primary and secondary labels are passed, we need to apply minor changes to the styling. Check below in styled component for more details.
+    const hasSecondaryLabel = labels && labels.props && labels.props.children && Children.count(labels.props.children) === 2;
```

Comment:
I am already facing a scenario where this is creating an issue. It would be better to let the user decide.
<img width="290" alt="Screenshot 2023-01-17 at 12 58 52 AM" src="https://user-images.githubusercontent.com/16203333/212754693-1938e44c-b59e-44b0-8aef-62fb69d4097d.png">

Reply:
It wasn't there in the design docs, so it got missed

Reply:
I have to agree with @AshishDhama here. We should let the user decide what to put into that label section.